src/overlays/actors/ovl_Obj_Raillift/z_obj_raillift.c

```diff
@@ -21,30 +43,242 @@ const ActorInit Obj_Raillift_InitVars = {
+    s32 pad;
+
+    if (OBJRAILLIFT_HAS_FLAG(&this->dyna.actor)) {
+        if (!Flags_GetSwitch(globalCtx, OBJRAILLIFT_GET_FLAG(&this->dyna.actor))) {
+            this->actionFunc = ObjRaillift_Idle;
+            return;
+        }
+
+        if (OBJRAILLIFT_GET_TYPE(&this->dyna.actor) == DEKU_FLOWER_PLATFORM) {
+            func_800B9010(&this->dyna.actor, NA_SE_EV_PLATE_LIFT_LEVEL - SFX_FLAG);
+        }
+    }
+
+    Math_Vec3s_ToVec3f(&nextPoint, &(&this->points[this->curPoint])[this->direction]);
+    Math_Vec3f_Diff(&nextPoint, &this->dyna.actor.world.pos, &this->dyna.actor.velocity);
+    speed = Math3D_Vec3fMagnitude(&this->dyna.actor.velocity);
+    if ((speed < (this->speed * 8.0f)) && (this->speed > 2.0f)) {
+        target = ((this->speed - 2.0f) * 0.1f) + 2.0f;
+        step = this->speed * 0.03f;
+    } else {
+        target = this->speed;
+        step = this->speed * 0.16f;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/ObjRaillift_Init.asm")
+    Math_StepToF(&this->dyna.actor.speedXZ, target, step);
+    if ((this->dyna.actor.speedXZ + 0.05f) < speed) {
+        Math_Vec3f_Scale(&this->dyna.actor.velocity, this->dyna.actor.speedXZ / speed);
+        this->dyna.actor.world.pos.x += this->dyna.actor.velocity.x;
+        this->dyna.actor.world.pos.y += this->dyna.actor.velocity.y;
+        this->dyna.actor.world.pos.z += this->dyna.actor.velocity.z;
+    } else {
+        this->curPoint += this->direction;
+        if (1) {}
+        this->dyna.actor.speedXZ *= 0.4f;
+        shouldTeleport = OBJRAILLIFT_SHOULD_TELEPORT(&this->dyna.actor);
+        updatePos = true;
+        if (((this->curPoint >= this->endPoint) && (this->direction > 0)) || ((this->curPoint <= 0) && (this->direction < 0))) {
+            if (!shouldTeleport) {
+                this->direction = -this->direction;
+                this->waitTimer = 10;
+                this->actionFunc = ObjRaillift_Wait;
+            } else {
+                endPoint = &this->points[this->endPoint];
+                this->curPoint = (this->direction > 0) ? (0) : (this->endPoint);
+                initialPoint = &this->points[0];
+                if (((initialPoint->x != endPoint->x) || (initialPoint->y != endPoint->y)) || (initialPoint->z != endPoint->z)) {
+                    this->actionFunc = ObjRaillift_Teleport;
+                    func_800C62BC(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+                    updatePos = false;
+                }
+            }
+        }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/ObjRaillift_Destroy.asm")
+        if (updatePos) {
+            ObjRaillift_UpdatePosition(this, this->curPoint);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Obj_Raillift_0x80A19910/func_80A19B98.asm")
+void ObjRaillift_Teleport(ObjRaillift* this, GlobalContext* globalCtx) {
+    // will teleport to what ever curpoint is set to
```

Comment:
```suggestion
/*
 * Will teleport to what ever curpoint is set to
 */
void ObjRaillift_Teleport(ObjRaillift* this, GlobalContext* globalCtx) {
```
If a comment is describing a function, then I think having like this is more standard